testing/process/test_forkedfunc.py

```diff
@@ -22,10 +22,10 @@ def raise_oserror():
         # BlockingIOError on py3k
         raise OSError("Resource temporarily unavailable")
     monkeypatch.setattr(os, "fork", raise_oserror)
-    with pytest.raises(OSError, "Resource temporarily unavailable"):
-        ff = py.process.ForkedFunc(boxf1)
-    ff.__del__()
-    assert not ff.tempdir.check()
+    with pytest.raises(OSError, match="Resource temporarily unavailable"):
+        py.process.ForkedFunc(boxf1)
+        # The second exception would be raised while leaving the with block
```

Comment:
to really trigger it a gc.collect call is needed in addition

Reply:
Added call to `gc.collect` outside the context manager.